Rename path type to recursive relationship (#217)
kuzudb/kuzu-docs: src/content/docs/cypher/query-clauses/match.md

--------------------------------------------------------------------
```

## Match recursive relationships

> You can also find paths that are of variable length between node records. Variable-length relationships

Reviewer: ```suggestion
You can also find recursive relationships (that are of variable length) between node records.
```

> are sometimes known as recursive patterns or recursive joins. Specifically, you can find variable-hop

Reviewer: ```suggestion
Specifically, you can find variable-length
```

- Karissa is found through `Noura <- User -> Karissa`
- Kitchener is found through `Noura <- User -> Kitchener`

### Return recursive relationships

> A recursive relationship has logical data type `RECURSIVE RELATIONSHIP` and is physically represented as `STURCT{LIST[NODE], LIST[RELATIONSHIP]}`. Returning a recursive relationship will return all properties 

Reviewer: ```suggestion
A recursive relationship has the logical data type `RECURSIVE_REL` and is physically represented as `STURCT{LIST[NODE], LIST[REL]}`. Returning a recursive relationship will return all properties 
```

----------------------------------------------------------------------------------------------
```

#### Filter recursive relationships
We also support running predicates on recursive relationships to constrain the relationship being traversed.

> The following query finds name of users and the number of path that are followed between 1 - 2 hops from Adam by person with age more than 45 and before 2022.

Reviewer: ```suggestion
The following query finds name of users and the number of path that are followed between 1-2 hops from Adam by person with age more than 45 and before 2022.
```

```

### Extracting nodes and relationships from a path

> Named path has logical data type [`RECURSIVE RELATIONSHIP`](/cypher/data-types/#path). Users can access nodes and relationships within a path through `nodes(p)` and `rels(p)` function calls.

Reviewer: ```suggestion
A named path has the logical data type [`RECURSIVE_REL`](/cypher/data-types/#recursive-rel). You can access nodes and relationships within a named path through `nodes(p)` and `rels(p)` function calls.
```
